### PR TITLE
APPS-1344 forwarding nextflow log to job monitor

### PR DESCRIPTION
**Pull request by YuxinShi0423**
Inspired by https://github.com/dnanexus/nucleus/blob/4f1ddb9adce22de30604cf78b923186efd30e3b3/execserver/worker/python3_rootfs/usr/bin/dx-exec-helper

### CODE REVIEW

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 1]
```diff
@@ -62,6 +81,17 @@ main() {
     echo "============================================================="
 
     filtered_inputs=""
+
+    # initiate log file and start forwarding it to the job monitor 
+    touch $LOG_NAME
+    if [[ $debug ]]; then
+      (while true; do truncate --size="<5G" $LOG_NAME; sleep 10; done) &
```
Comment (mhrvol) [inline src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:88]: 5G is pretty huge, I think the log size is capped at 2MB or 100k lines.

Comment (YuxinShi0423) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:88]: https://github.com/dnanexus/nucleus/blob/4586131b460dd059bfe6a6c187698c81db2b0224/execserver/scripts/relay_job_logging.py#L179

I saw we are using 2MB or 100 lines as the limit. Will adopt those configs. Thanks @mhrvol !

Comment (YuxinShi0423) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh]: Refactored.

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 2]
```diff
@@ -34,8 +53,9 @@ on_exit() {
 main() {
     set -f
 
-    [[ $debug ]] && set -x && env | sort
     [[ $debug ]] && export NXF_DEBUG=2
+    [[ $debug ]] && TRACE_CMD="-trace nextflow.plugin"
```
Comment (mhrvol) [inline src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:57]: This seems to run the debug commands even if debug=false

Could we change this to 
    if $debug ; then
      TRACE_CMD="-trace nextflow.plugin"
      set -x && env | sort
      export NXF_DEBUG=2
    fi

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 3]
```diff
@@ -64,7 +84,18 @@ main() {
     filtered_inputs=()
 
     @@RUN_INPUTS@@
-    nextflow -trace nextflow.plugin $nf_advanced_opts -log ${LOG_NAME} run @@RESOURCES_SUBPATH@@ @@PROFILE_ARG@@ -name run-${NXF_UUID} $nf_run_args_and_pipeline_params "${filtered_inputs[@]}"
+    nextflow -trace nextflow.plugin $nf_advanced_opts -log ${LOG_NAME} run @@RESOURCES_SUBPATH@@ @@PROFILE_ARG@@ -name run-${NXF_UUID} $nf_run_args_and_pipeline_params "${filtered_inputs[@]}" & NXF_EXEC_PID=$!
```
Comment (mhrvol) [inline src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:87]: Should we use nextflow "${TRACE_CMD}" instead of hardcoding the nextflow.plugin?

Comment (YuxinShi0423) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh]: sorry it got overwritten when I merge the dev. Fixed. Thanks Marek!

**Comment by mhrvol** [comment]
There is one conflict with the nextflow-dev. Otherwise this looks good!

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh (hunk 2 above)
Comment (YuxinShi0423) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh]: Fixed. Thanks!

**Review by mhrvol — APPROVED** [review]

**Pull request merged**